Libplanet/Blocks/BlockCommitExtensions.cs

```diff
@@ -0,0 +1,80 @@
+        /// <see cref="VoteFlag.Commit"/> of <see cref="BlockCommit.BlockHash"/>, otherwise returns
+        /// <see langword="false"/>.</returns>
+        public static bool HasTwoThirdCommits(
+            this BlockCommit commit,
+            IEnumerable<PublicKey> validators) =>
+            validators.Count() * 2 / 3 < commit.Votes.Count(vote =>
+                vote.Flag == VoteFlag.Commit && vote.BlockHash.Equals(commit.BlockHash));
+
+        /// <summary>
+        /// Checks whether <see cref="BlockCommit"/> has same validator public key with given
+        /// <paramref name="validators"/>.
+        /// </summary>
+        /// <param name="commit">A <see cref="BlockCommit"/> to check.</param>
+        /// <param name="validators">A <see cref="IEnumerable{T}"/> of validator public key.</param>
+        /// <returns>Returns <see langword="true"/> if <see cref="BlockCommit"/> has same validator
+        /// public keys with <paramref name="validators"/>, otherwise returns
+        /// <see langword="false"/>. </returns>
+        public static bool HasSameValidators(
+            this BlockCommit commit,
+            IEnumerable<PublicKey> validators)
+        {
+            List<PublicKey> iaVoteToHashSetList = new List<PublicKey>();
+            foreach (var key in commit.Votes.Select(x => x.Validator))
+            {
+                iaVoteToHashSetList.Add(key);
+            }
+
+            ImmutableHashSet<PublicKey> voteValidators = iaVoteToHashSetList.ToImmutableHashSet();
+            var publicKeys = validators as PublicKey[] ?? validators.ToArray();
+            ImmutableHashSet<PublicKey> previousValidators = publicKeys.ToImmutableHashSet();
+
+            return iaVoteToHashSetList.Count == publicKeys.Count() &&
+                   voteValidators.SetEquals(previousValidators);
+        }
+
+        /// <summary>
+        /// Checks whether <see cref="BlockCommit"/> has valid signatures for every votes.
+        /// </summary>
+        /// <param name="commit">A <see cref="BlockCommit"/> to check.</param>
+        /// <returns>Returns <see langword="true"/> if <see cref="BlockCommit.Votes"/> has valid
+        /// signatures, otherwise return <see langword="false"/>.</returns>
+        /// <remarks>A vote with <see cref="VoteFlag.Null"/> or <see cref="VoteFlag.Unknown"/> is
+        /// excluded from target.</remarks>
+        public static bool HasValidVotes(this BlockCommit commit) =>
+            commit.Votes.All(
+                vote =>
+                {
+                    // If the flag of a vote is not null or unknown, it should have valid signature.
+                    if (vote.Verify())
+                    {
+                        return true;
+                    }
+
+                    if (vote.Flag == VoteFlag.Null || vote.Flag == VoteFlag.Unknown)
+                    {
+                        return true;
+                    }
+
+                    return false;
+                });
```

Comment:
I think this can be compressed to a single line.

Reply:
Amended.
https://github.com/planetarium/libplanet/blob/15f67c376b0bade3c0a32b0ed3475d42fd44f160/Libplanet/Blocks/BlockCommitExtensions.cs#L54-L65